src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/measure/ActionMeterTest.kt

```diff
@@ -158,6 +160,39 @@ class ActionMeterTest {
         )
     }
 
+    @Test
+    fun shouldHookAction() {
+        val output = CollectionActionMetricOutput(mutableListOf())
+        val countingPostMeasurementAction = CountingPostMeasurementAction()
+        val tick = ofSeconds(1)
+        val clock = TickingClock(start, tick)
+        val actionMeter = ActionMeter.Builder(
+            output = output
+        )
+            .virtualUser(vu)
+            .clock(clock)
+            .postMeasurementAction(countingPostMeasurementAction)
+            .build()
+
+        actionMeter.measure(CREATE_ISSUE, clock::tick)
+        actionMeter.measure(VIEW_BOARD, clock::tick)
+
+        assertThat(countingPostMeasurementAction.getCount()).isEqualTo(2)
+        assertThat(output.metrics.map { it.duration }).hasOnlyOneElementSatisfying { ofSeconds(1) }
+    }
+
+    private class CountingPostMeasurementAction : PostMeasurementAction {
```

Comment:
`Action` has existing connotation in `jira-actions`. I'd call it directly: `PostMeasurementHook` or `PostMetricHook`.